src/main/java/dev/espi/ebackup/BackupUtil.java

```diff
@@ -142,22 +142,29 @@ public static void doBackup(boolean uploadToServer) {
 
             // upload to ftp/sftp
             if (uploadToServer && eBackup.getPlugin().ftpEnable) {
-                File f = new File(eBackup.getPlugin().backupPath + "/" + fileName + ".zip");
-                if (eBackup.getPlugin().ftpType.equals("sftp")) {
-                    eBackup.getPlugin().getLogger().info("Uploading backup to SFTP server...");
-                    uploadSFTP(f);
-                } else if (eBackup.getPlugin().ftpType.equals("ftp")) {
-                    eBackup.getPlugin().getLogger().info("Uploading backup to FTP server...");
-                    uploadFTP(f);
-                }
-
-                // if the upload is able to go smoothly, delete local backup
-                if (eBackup.getPlugin().deleteAfterUpload) {
-                    if (f.delete()) {
-                        eBackup.getPlugin().getLogger().info("Successfully deleted local backup zip after upload.");
-                    } else {
-                        eBackup.getPlugin().getLogger().warning("Unable to delete local backup zip after upload.");
+                if (!eBackup.getPlugin().isInUpload.get()) {
+                    File f = new File(eBackup.getPlugin().backupPath + "/" + fileName + ".zip");
+                    if (eBackup.getPlugin().ftpType.equals("sftp")) {
+                        eBackup.getPlugin().getLogger().info("Uploading " + fileName + " to SFTP server...");
+                        Bukkit.getScheduler().runTaskAsynchronously(eBackup.getPlugin(), () -> {
+                            try {
```

Comment:
I would recommend moving the `eBackup.getPlugin().isInUpload.set(true);` here, and move `eBackup.getPlugin().isInUpload.set(false);` into a `finally` block. This is so that we ensure `isInUpload` remains false even if an exception occurred.

Reply:
I agree with you.Thanks for advice. :)